tensornetwork/network_operations.py

```diff
@@ -817,3 +833,102 @@ def get_neighbors(node: AbstractNode) -> List[AbstractNode]:
         neighbors.append(edge.node1)
         neighbors_set.add(edge.node1)
   return neighbors
+
+
+JSON_TYPE_DICT = {
+    Node: 'Node',
+}
+
+
+def nodes_to_json(nodes: List[AbstractNode]) -> str:
+  """
+  Create a JSON string representing the Tensor Network made up of the given 
+  nodes. Nodes and their attributes, edges and their attributes and tensor
+  values are included.
+  
+  Tensors are serialized according the the format used by each tensors backend.
+  
+  For edges spanning included nodes and excluded nodes the edge attributes are 
+  preserved in the serialization but the connection to the excluded node is 
+  dropped. The original edge is not modified.
+  
+  Args:
+    nodes: A list of nodes making up a tensor network.
+    
+  Returns:
+    A string representing the JSON serialized tensor network.
+    
+  Raises:
+    ValueError: If a node does not have a serialized 'type' string.
+  """
+  type_dict = JSON_TYPE_DICT
+  network_dict = {
+      'nodes': [],
+      'edges': [],
+  }
+  node_id_dict = {}
+
+  for i, node in enumerate(nodes):
+    t = type_dict.get(type(node), None)
+    if not t:
+      ValueError(f'Unsupported serialization type: {type(node)}')
+
+    node_id_dict[node] = i
+    network_dict['nodes'].append({
+        'id': i,
+        'type': t,
```

Comment:
Go ahead and remove this type stuff.

Reply:
I had left it in case we want to have a mechanism in the future for other node classes to be added and serialized. I can remove if you think that is not necessary. @Thenerdstation